var/spack/repos/builtin/packages/go/package.py

```diff
@@ -128,12 +127,7 @@ class Go(Package):
     provides('golang')
 
     depends_on('git', type=('build', 'link', 'run'))
-    # TODO: Make non-c self-hosting compilers feasible without backflips
-    # should be a dep on external go compiler
-    if platform.machine() == 'aarch64':
-        depends_on('gcc languages=go', type='build')
-    else:
-        depends_on('go-bootstrap', type='build')
+    depends_on('go-bootstrap', type='build')
```

Comment:
This, combined with the `go-bootstrap` changes, looks like it makes building for `aarch64` impossible.

Reply:
Yes, as far as I can tell it's impossible to build go on aarch64. I'm now using an external from homebrew.

Reply:
It can't build from `gcc languages=go`?

Reply:
Not at present. Currently GCC12 is required to build on M1 (and it's not yet released and therefore unavailable through spack); and the homebrew installation doesn't include `go` support. Even worse,  when I tried to force `gcc languages=go` as an alternative dependency,  https://github.com/spack/spack/issues/28827 reared its ugly head and made my non-go external "install" with false support for `go`. And even then, I believe additional [external variables are necessary](https://github.com/golang/go/issues/47771) are necessary to get the GCC bootstrap to work, and `GCCGO` isn't currently handled in the `go` bootstrap mechanism... so there's clearly work to be done here. Disabling `go` on Apple Silicon at least reflects the current state of affairs rather than leading into a bizarre rabbit hole  of missing features.

Reply:
That's painful.  Probably best somewhere else, but maybe we should add an escape hatch to register an external go as a bootstrapper.  I'd been hoping to wait until the go package could optionally depend on an older version of itself though.